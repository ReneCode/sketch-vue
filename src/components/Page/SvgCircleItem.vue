<template>
  <circle :cx="item.svg.x" :cy="item.svg.y" :r="item.svg.radius" :style="style">
  </circle>
</template>

<script>
export default {
  name: 'svg-circle-item',
  props: ['item'],
  computed: {
    style() {
      let style = {
      };
      if (this.item.svg.stroke) {
        style.stroke = this.item.svg.stroke;
      }
      if (this.item.svg.strokeWidth) {
        style.strokeWidth = this.item.svg.strokeWidth;
      }
      if (this.item.svg.fill) {
        style.fill = this.item.svg.fill;
      }
      return style;
    }
  }
}
</script>

<style>
circle {
  stroke: #630;
  stroke-width: 2px;
  fill: none;
}
</style>
