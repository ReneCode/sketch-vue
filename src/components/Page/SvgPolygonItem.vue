<template>
  <polygon :points="points" :style="style">
  </polygon>
</template>

<script>
export default {
  name: "svg-polygon-item",
  props: ["item"],
  computed: {
    style() {
      let style = {};
      if (this.item.svg.stroke) {
        style.stroke = this.item.svg.stroke;
      }
      if (this.item.svg.strokeWidth) {
        style.strokeWidth = this.item.svg.strokeWidth;
      }
      if (this.item.svg.fill) {
        style.fill = this.item.svg.fill;
      }
      return style;
    },

    points() {
      let str;
      for (let pt of this.item.svg.points) {
        if (!str) {
          str = `${pt.x},${pt.y}`;
        } else {
          str += `,${pt.x},${pt.y}`;
        }
      }
      return str;
    }
  }
};
</script>

<style>
polygon {
  fill: none;
}
</style>

