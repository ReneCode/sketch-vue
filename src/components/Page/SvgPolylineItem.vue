<template>
  <polyline :points="points" :style="style">
  </polyline>
</template>

<script>
export default {
  name: 'svg-polyline-item',
  props: ['item'],
  computed: {
    style() {
      let style = {};
      if (this.item.svg.fill) {
        style.fill = this.item.svg.fill;
      }
      if (this.item.svg.stroke) {
        style.stroke = this.item.svg.stroke;
      }
      if (this.item.svg.strokeWidth) {
        style.strokeWidth = this.item.svg.strokeWidth;
      }
      return style;
    },
    points() {
      let str;
      for (let pt of this.item.svg.points) {
        if (!str) {
          str = `${pt.x},${pt.y}`;
        } else {
          str += `,${pt.x},${pt.y}`;
        }
      }
      return str;
    }
  }
}
</script>

<style>
polyline {
  fill: none;
}
</style>