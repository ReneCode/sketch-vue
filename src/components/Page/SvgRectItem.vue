<template>
  <rect :x="item.svg.x" :y="item.svg.y" :width="item.svg.width" :height="item.svg.height" :style="style">
  </rect>
</template>

<script>
export default {
  name: 'svg-rect-item',
  props: ['item'],
  computed: {
    style() {
      let style = {
      };
      if (this.item.svg.stroke) {
        style.stroke = this.item.svg.stroke;
      }
      if (this.item.svg.strokeWidth) {
        style.strokeWidth = this.item.svg.strokeWidth;
      }
      if (this.item.svg.fill) {
        style.fill = this.item.svg.fill;
      }
      return style;
    }
  }
}
</script>

<style>
rect {
  fill: none;
}
</style>
