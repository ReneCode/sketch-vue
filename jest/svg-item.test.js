import { mount } from 'vue-test-utils'
import SvgRectItem from '../src/components/Page/SvgRectItem'
import SvgCircleItem from '../src/components/Page/SvgCircleItem'
import SvgPolygonItem from '../src/components/Page/SvgPolygonItem'
import SvgPolylineItem from '../src/components/Page/SvgPolylineItem'

import ItemRectangle from '../src/models/item-rectangle'
import ItemCircle from '../src/models/item-circle'
import ItemPolygon from '../src/models/item-polygon'
import ItemPolyline from '../src/models/item-polyline'

import Point from '../src/models/point'

describe('SvgRectItem', () => {

  const p1 = new Point(33, 44);
  const p2 = new Point(55, 77);
  const itemRect = new ItemRectangle(p1, p2);
  itemRect.svg.stroke = "red";
  itemRect.svg.strokeWidth = "3px";
  itemRect.svg.fill = "blue";

  const wrapper = mount(SvgRectItem, {
    propsData: {
      item: itemRect
    }
  })

  it('renders the markup', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SvgCircleItem', () => {
  const pt = new Point(50, 70);
  const item = new ItemCircle(pt, 40);
  item.svg.stroke = "red";
  item.svg.strokeWidth = "3px";
  item.svg.fill = "blue";

  const wrapper = mount(SvgCircleItem, {
    propsData: {
      item: item
    }
  })

  it('renders the markup', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SvgPolygonItem', () => {
  const item = new ItemPolygon();
  item.addPoint(new Point(10, 20));
  item.addPoint(new Point(30, 5));
  item.addPoint(new Point(60, 50));
  item.svg.stroke = "red";
  item.svg.strokeWidth = "3px";
  item.svg.fill = "blue";

  const wrapper = mount(SvgPolygonItem, {
    propsData: {
      item: item
    }
  })

  it('renders the markup', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SvgPolylineItem', () => {
  const item = new ItemPolyline();
  item.addPoint(new Point(10, 20));
  item.addPoint(new Point(30, 5));
  item.addPoint(new Point(60, 50));
  item.svg.stroke = "red";
  item.svg.strokeWidth = "3px";
  item.svg.fill = "blue";

  const wrapper = mount(SvgPolylineItem, {
    propsData: {
      item: item
    }
  })

  it('renders the markup', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
