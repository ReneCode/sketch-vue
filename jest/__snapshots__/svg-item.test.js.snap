// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgCircleItem renders the markup 1`] = `<circle cx="50" cy="70" r="40" style="stroke: red; stroke-width: 3px; fill: blue;"></circle>`;

exports[`SvgPolygonItem renders the markup 1`] = `<polygon points="10,20,30,5,60,50" style="stroke: red; stroke-width: 3px; fill: blue;"></polygon>`;

exports[`SvgPolylineItem renders the markup 1`] = `<polyline points="10,20,30,5,60,50" style="fill: blue; stroke: red; stroke-width: 3px;"></polyline>`;

exports[`SvgRectItem renders the markup 1`] = `<rect x="33" y="44" width="22" height="33" style="stroke: red; stroke-width: 3px; fill: blue;"></rect>`;
